Fixed nativeq (Q) Switching Function Analytical Derivatives

nm3787 opened the pull request:
<!--
  Feel free to delete not relevant sections below
-->

##### Description

<!-- describe here your pull request -->
This fixes the derivative for the nativeq (Q) switching function in tools/SwitchingFunction.cpp. The derivative of the Q switching function should be multiplied by beta according to the chain rule and not divided by r0. This would affect the forces acting on CVs using this switching function. This modification was discussed with @hockyg.

##### Type of contribution

<!--
  Please select the type of your contribution among these:
  (Change [ ] to [X] to tick an option)
-->
- [X] changes to code or doc authored by PLUMED developers, or additions of code in the core or within the default modules

##### Copyright

<!--
  In case you picked one of the first two choices
  MAKE SURE TO TICK ALSO THE FOLLOWING BOX
-->

- [X] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

##### Tests

<!--
  Make sure these boxes are checked. For Travis-CI tests, you can wait for them
  to be completed monitoring this page after your pull request has been submitted:
  http://travis-ci.org/plumed/plumed2/pull_requests
-->

- [X] I added a new regtest or modified an existing regtest to validate my changes.
- [X] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

<!--
  After your branch has been merged to the desired branch and then to plumed2/master, and after the
  plumed official manual has been updated, please check out the coverage scan at
  http://www.plumed.org/coverage-master
  In case your new features are not well covered, please try to add more complete regtests.
-->

carlocamilloni commented on src/tools/SwitchingFunction.cpp:
```diff
@@ -487,7 +487,7 @@ double SwitchingFunction::calculate(double distance,double&dfunc)const {
     } else plumed_merror("Unknown switching function type");
 // this is for the chain rule:
-    dfunc*=invr0;
+    if(type!=nativeq) dfunc*=invr0;
```
Instead than this if I would divide for invr0 on line 454
if you can change this I will then merge it, thank!

carlocamilloni commented:
@nm3787 thanks for reporting and fixing, if you can change the code as suggested I will then merge it

hockyg commented:
> @nm3787 thanks for reporting and fixing, if you can change the code as suggested I will then merge it

Hi @carlocamilloni - I see why this makes sense from an efficiency point of view, but maybe you can add a code comment to warn people in case they add future switching functions?

nm3787 commented:
> @nm3787 thanks for reporting and fixing, if you can change the code as suggested I will then merge it

Thank you @carlocamilloni for the great suggestion. I have implemented it and added a comment as @hockyg suggested.
